cmd/thanos/rule.go

```diff
@@ -934,3 +948,34 @@ func reloadRules(logger log.Logger,
 	}
 	return errs.Err()
 }
+
+func tableLinkForExpression(tmpl string, expr string) (string, error) {
+	// template example: "/graph?g0.expr={{.Expr}}&g0.tab=1"
+	escapedExpression := url.QueryEscape(expr)
+
+	escapedExpr := Expression{Expr: escapedExpression}
+	t, err := template.New("url").Parse(tmpl)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse template")
+	}
+
+	var buf bytes.Buffer
+	if err := t.Execute(&buf, escapedExpr); err != nil {
+		return "", errors.Wrap(err, "failed to execute template")
+	}
+	return buf.String(), nil
+}
+
+func validateTemplate(tmplStr string, data Expression) error {
```

Comment:
What is the point of the second function parameter if we always provide just the test expression? Maybe we should just provide it directly in `tmpl.Exectue`?

Reply:
You are right, there is no need for validation. Just removed it.

@matej-g Could you please take another look?